pkg/gateway/model_build_servicenetwork.go

```diff
@@ -63,25 +70,29 @@ func (t *serviceNetworkModelBuildTask) buildModel(ctx context.Context) error {
 
 func (t *serviceNetworkModelBuildTask) buildServiceNetwork(ctx context.Context) error {
 	spec := latticemodel.ServiceNetworkSpec{
-		Name:           t.gateway.Name,
-		Namespace:      t.gateway.Namespace,
-		Account:        config.AccountID,
-		AssociateToVPC: false,
+		Name:      t.gateway.Name,
+		Namespace: t.gateway.Namespace,
+		Account:   config.AccountID,
 	}
 
-	// by default it is true
-	spec.AssociateToVPC = true
+	// default associateToVPC is true
+	associateToVPC := true
 	if len(t.gateway.ObjectMeta.Annotations) > 0 {
 		if value, exist := t.gateway.Annotations[LatticeVPCAssociationAnnotation]; exist {
-			if value == "true" {
-				spec.AssociateToVPC = true
-			} else {
-				spec.AssociateToVPC = false
-			}
+			associateToVPC = value == "true"
+		}
+	}
 
+	if t.vpcAssociationPolicy != nil {
+		if t.vpcAssociationPolicy.Spec.AssociateWithVpc != nil {
```

Comment:
did you set default value as true in the new CRD?

Reply:
No, In CRD, by default it's empty https://github.com/aws/aws-application-networking-k8s/blob/520448301016d569a458394b05ebc9e94e4f2ffb/pkg/apis/applicationnetworking/v1alpha1/vpcassociationpolicy_types.go#L58

In the controller for both old and new logic,  by default it is true

Reply:
have you considered setting default value via kubebuilder annotation?